waspls/src/Wasp/LSP/Handlers.hs

```diff
@@ -0,0 +1,118 @@
+module Wasp.LSP.Handlers
+  ( initializedHandler,
+    didOpenHandler,
+    didChangeHandler,
+    didSaveHandler,
+  )
+where
+
+import Control.Lens ((+~), (^.))
+import Control.Monad.Trans (lift)
+import Control.Monad.Trans.Except (catchE, throwE)
+import Data.Function ((&))
+import Data.Text (Text)
+import qualified Data.Text as T
+import Language.LSP.Server (Handlers, LspT)
+import qualified Language.LSP.Server as LSP
+import Language.LSP.Types
+import Language.LSP.Types.Lens
+import Language.LSP.VFS (virtualFileText)
+import qualified Wasp.Analyzer
+import qualified Wasp.Analyzer.AnalyzeError as WE
+import Wasp.Analyzer.Parser (Ctx (Ctx))
+import Wasp.Analyzer.Parser.SourceRegion (getRgnEnd, getRgnStart)
+import Wasp.LSP.State (HandlerM, ServerConfig, Severity (..), State)
+
+liftLSP :: LspT ServerConfig IO a -> HandlerM a
+liftLSP m = lift (lift m)
+
+readUri :: Uri -> HandlerM Text
+readUri _uri = do
```

Comment:
What does this function do? Reading it implementation I see it is very focused on dealing with LSPs virtual file system, while the name sounds like it just reads some general URI in the context of HandleM monad -> should we make the name more explicit as to what it does?